### PR TITLE
Add graceful shutdown

### PR DESCRIPTION
**Pull request by jason-fox**
When a shutdown signal is sent, we need to do graceful shutdown out of Node.js and Express to avoid side effects
like finishing active requests before closing server, clean up resources, db connections etc.

see: https://stackfame.com/node-express-graceful-shutdown

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,6 +22,7 @@
     "clean": "rm -rf package-lock.json && rm -rf node_modules && rm -rf coverage && rm -rf .nyc_output",
     "healthcheck": "node ./bin/healthcheck",
     "lint": "eslint lib/ bin/iota-manager test/ --cache --fix",
+    "start": "node ./bin/iotagent-manager",
```
Comment (fgalan) [inline package.json:25]: I think npm targets are described here in this repo: https://github.com/telefonicaid/iotagent-manager#-development-documentation

Comment (jason-fox) [reply package.json:25]: Fixed 4ba0df9

File: README.md [hunk 2]
```diff
@@ -181,43 +206,52 @@ IoTAgent Manager with the same protocol and resource of an already existing agen
...
+
 ```bash
 npm run
 ```
 
 The following sections show the available options in detail.
 
+### Start
+
+Runs a local version of the IoT Agent
```
Comment (fgalan) [inline README.md:247]:
```suggestion
Runs a local version of the IoT Manager
```

File: README.md [hunk 3]
```diff
@@ -4,118 +4,138 @@
...
-have the application running. The Docker image will automatically start.
 
-In order to run the docker image, first you must have a MongoDB instance running. You can achieve this by executing
-the followin command:
-```
+The Docker automatically starts listening in the API ports, so there is no need to execute any process in order to have
+the application running. The Docker image will automatically start.
+
+In order to run the docker image, first you must have a MongoDB instance running. You can achieve this by executing the
+followin command:
```
Comment (fgalan) [inline README.md:106]: Typo:

```suggestion
following command:
```

File: README.md (hunk 2 above)
Comment (fgalan) [reply README.md:247]: It seems this is the only thing pending.

**Review by fgalan — APPROVED** [review]
LGTM... although autolinter changes didn't make the review easy ;)

**Pull request merged**